expo/tests/scheduler/test_schedule_acts.py

```diff
@@ -0,0 +1,203 @@
+        self.assert_good_form_display(response)
+
+    def test_good_user_post_success(self):
+        login_as(self.privileged_profile, self)
+        response = self.client.post(
+            self.url,
+            data=self.get_basic_post())
+        self.assertRedirects(
+            response,
+            reverse('home', urlconf='gbe.urls'))
+        self.assertNotEqual(self.context.acts[0].title,
+                            'changed title')
+        self.assertNotEqual(str(self.context.acts[0].performer),
+                            'changed performer')
+
+    def test_good_user_post_success_w_label(self):
+        self.context.order_act(self.context.acts[0], 2)
+        login_as(self.privileged_profile, self)
+        response = self.client.post(
+            self.url,
+            data=self.get_basic_post())
+        self.assertRedirects(
+            response,
+            reverse('home', urlconf='gbe.urls'))
+        allocation = self.context.sched_event.resources_allocated.filter(
+            resource__actresource___item=self.context.acts[0]).first()
+        self.assertEqual(allocation.ordering.order, 1)
+
+    def test_good_user_get_success_w_label(self):
+        self.context.order_act(self.context.acts[0], 2)
+        login_as(self.privileged_profile, self)
+        response = self.client.get(
+            self.url)
+        self.assert_good_form_display(response)
+        allocation = self.context.sched_event.resources_allocated.filter(
+            resource__actresource___item=self.context.acts[0]).first()
+        self.assertContains(
+            response,
+            '<input id="id_allocation_' + str(allocation.pk) +
+            '-order" name="allocation_' + str(allocation.pk) +
+            '-order" type="number" value="2" />')
+
+    def test_good_user_post_invalid(self):
+        login_as(self.privileged_profile, self)
+        allocation = self.context.sched_event.resources_allocated.filter(
+            resource__actresource___item=self.context.acts[0]).first()
+        data = self.get_basic_post()
+        data['allocation_' + str(allocation.pk) +
+             '-show'] = 'bad'
+        data['allocation_' + str(allocation.pk) + '-order'] = 'very bad'
+        data['allocation_' + str(allocation.pk) + '-actresource'] = \
+            'adfasdfasdfkljasdfklajsdflkjasdlkfjalksjdflkasjdflkjasdl'
+        response = self.client.post(
+            self.url,
+            data=data)
+        self.assertRedirects(
+            response,
+            reverse('home', urlconf='gbe.urls'))
+
+    def test_good_user_change_show(self):
+        new_show = ShowContext(conference=self.context.conference)
+        login_as(self.privileged_profile, self)
+        allocation = self.context.sched_event.resources_allocated.filter(
+            resource__actresource___item=self.context.acts[0]).first()
+        data = self.get_basic_post()
+        data['allocation_' + str(allocation.pk) +
```

Comment:
again, please do string formatting in the canonical way (several other instances in this commit)